### PR TITLE
Add nn.functional.kl_div

### PR DESCRIPTION
**Pull request by marigoold**
背景：espnet2 需求
与此同时发现文档中说 reduction 可以是 batchmean，但是代码中不支持，这里增加了对应的支持和 torch 对齐

### CODE REVIEW

File: python/oneflow/nn/modules/loss.py [hunk 1]
```diff
@@ -450,7 +450,12 @@ class KLDivLoss(_Loss):
     """
 
     def __init__(self, reduction: str = "mean", log_target: bool = False) -> None:
-        super(KLDivLoss, self).__init__(reduction)
+        if reduction == "batchmean":
+            super(KLDivLoss, self).__init__("sum")
+            self.reduction = "batchmean"
+        else:
+            super(KLDivLoss, self).__init__(reduction)
+
```
Comment (marigoold) [inline python/oneflow/nn/modules/loss.py:458]: 这里传入 batchmean 会触发 assert，传了一个其他值来绕过 assert

**Review by BBuf — APPROVED** [review]
